ape_ganache/provider.py

```diff
@@ -321,6 +322,12 @@ def get_virtual_machine_error(self, exception: Exception) -> VirtualMachineError
 
         return ContractLogicError(revert_message=message)
 
+    def unlock_account(self, address: AddressType) -> bool:
+        self._make_request("evm_addAccount", [address, ""])
+        self._make_request("personal_unlockAccount", [address, "", 9999999999])
+        self._make_request("evm_setAccountBalance", [address, to_wei(1, "ether")])
```

Comment:
What concerns me here is that we mix unlocking of account with funding the account. Since I am not sure how the provider architecture is supposed to work, I can only say that it looks strange to combine the two actions into a single method like that.

@fubuloubu what's your opinion on that?

Reply:
I was also not sure about that, also i can say without this things will get complex in tests, if i put contract as sender directly, then i dont have any time to send some balance to the contract before this.

Reply:
Worth a try without it, it should work with zero fee on test chains. Also should just make it match the other providers, I don't think they give ETH when you unlock them

Reply:
discarded balance part